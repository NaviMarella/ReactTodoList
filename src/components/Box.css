
.box_container{
    margin: 5%;
}

.box_large_one{
    margin: 2% 10%;
}