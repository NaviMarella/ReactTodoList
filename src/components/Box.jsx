import React from "react";
import "./Box.css";

const Box = ({children, type, ...otherProps}) => {
    const styles = [
        'box_container',
        type === "large" ? "box_large_one": ""
    ].join(" ");
    return (
        <div className={styles} {...otherProps}>
            {children}
        </div>
    )
}

export default Box;