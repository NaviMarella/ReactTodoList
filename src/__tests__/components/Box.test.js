import React from 'react';
import { shallow } from 'enzyme';
import Box from './../../components/Box';

describe('Box Component', () => {
  test('render snapshot', () => {
      let props = {
        type: "large",
        children: <div>Test</div>
      }
    const wrapper = shallow(<Box {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('render snapshot with small size', () => {
      let props = {
        type: "small",
        children: <div>Test</div>
      }
    const wrapper = shallow(<Box {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
