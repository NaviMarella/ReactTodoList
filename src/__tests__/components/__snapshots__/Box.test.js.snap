// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Component render snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box
    type="large"
  >
    <div>
      Test
    </div>
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Test
      </div>,
      "className": "box_container box_large_one",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
      },
      "ref": null,
      "rendered": "Test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Test
        </div>,
        "className": "box_container box_large_one",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Box Component render snapshot with small size 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Box
    type="small"
  >
    <div>
      Test
    </div>
  </Box>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        Test
      </div>,
      "className": "box_container ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Test",
      },
      "ref": null,
      "rendered": "Test",
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          Test
        </div>,
        "className": "box_container ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Test",
        },
        "ref": null,
        "rendered": "Test",
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
